### PR TITLE
Fuse split rotary for Group Query Attention

### PR DESCRIPTION
**Pull request by jinhongyii**

### CODE REVIEW

**Comment by junrushao** [comment]
CC @masahi as the original author of this optimization :)

File: mlc_llm/transform/fuse_split_rotary_embedding.py [hunk 1]
```diff
@@ -70,10 +70,84 @@ def split_rotary(
 
     return split_rotary
 
+def get_split_rotary_group_query_attention(num_query_heads, num_kv_heads,  head_dim, position_embedding_base):
```
Comment (MasterJH5574) [inline mlc_llm/transform/fuse_split_rotary_embedding.py:73]: I’m wondering if we can merge the two `get_split_rotary` into one.

**Comment by masahi** [comment]
nice, does a big model that use MQA / GQA benefit from this pass? The speed up for 13B was much smaller than 7B when I tested.

**Comment by jinhongyii** [comment]
llama2 70b uses GQA. Indeed the speed up is small. Only 2% I observed

**Review by MasterJH5574 — APPROVED** [review]

**Pull request merged**